File: fizzbuzz.js
const COUNT_LIMIT = 100 ;

const COUNT_ARRAY = [] ;



function format (int) {  // int

	const isFizz = (int % 3) === 0 ;

	const isBuzz = (int % 5) === 0 ;

	if ( ! isFizz && ! isBuzz) {

		return `${int}` ;

	}

	if (isFizz && isBuzz) {

		return 'FizzBuzz' ;

	}

	if (isFizz) {

		return 'Fizz' ;

	}

	// else, is buzz
	return 'Buzz' ;

}



for (let i = 1 ; i <= COUNT_LIMIT ; i++) {

	const formattedResult = format (i) ;

	COUNT_ARRAY.push (formattedResult) ;

}



console.log (COUNT_ARRAY) ;

File: fizzbuzz.fs
variable count 1 count !
variable state 0 state !
variable limit 100 limit !

: testIsFizz dup 3 mod 0 = if 1 else 0 then state @ + state ! ;
: testIsBuzz dup 5 mod 0 = if 2 else 0 then state @ + state ! ;

: stateUpdate 0 state ! count @ testIsFizz testIsBuzz drop ;

: stateShow
	state @ 0 = if count @ . then
	state @ 1 = if ." fizz " then
	state @ 2 = if ." buzz " then
	state @ 3 = if ." fizzbuzz " then	
;

: testLimit count @ limit @ > invert ;
: incrementCount count @ 1 + count ! ;

: main
	begin
		stateUpdate
		stateShow
		incrementCount
		testLimit
	while
	repeat
;

main

